tensorbay/cli/log.py

```diff
@@ -189,36 +240,74 @@ def _sort(self) -> None:
         self._sorted_commits.insert(index, commits)
         self._keys.insert(index, date)
 
-    def _generate_commits(self) -> Iterator[Commit]:
-        """Get the latest commit in commit list.
+    def _generate_commits_and_drafts(self) -> Iterator[Tuple[Union[Commit, Draft], bool]]:
+        """Get the latest commit or draft.
 
         Yields:
-            The latest commit.
+            The latest commit or draft.
 
         """
         while True:
             try:
-                latest_commit = self._sorted_commits[-1].pop(0)
+                latest_commit = self._sorted_commits[-1][0]
             except IndexError:
+                if self._sorted_drafts:
+                    yield from product(self._sorted_drafts[::-1], [True])
                 return
-            if self._merge(latest_commit):
-                continue
-            yield latest_commit
-            self._sort()
+            latest_draft: Optional[Draft] = self._sorted_drafts[-1] if self._sorted_drafts else None
+            if latest_draft and latest_draft.updated_at > latest_commit.committer.date:
+                yield latest_draft, True
+                del self._sorted_drafts[-1]
+            else:
+                del self._sorted_commits[-1][0]
+                if self._merge(latest_commit):
+                    continue
+                yield latest_commit, False
+                self._sort()
 
-    def generate_commits(self) -> Iterator[str]:
-        """Get the the log of all commits.
+    def generate_commits_and_drafts(self) -> Iterator[str]:
+        """Get the the log of all commits and open drafts.
 
         Yields:
-            The commit log.
+            The commit or draft log.
 
         """
-        for commit in self._generate_commits():
-            commit_id = commit.commit_id
-            yield self._printer(commit, self._commit_id_to_branches.get(commit_id))
+        for element, is_draft in self._generate_commits_and_drafts():
+            if is_draft:
+                yield self._draft_printer(element)  # type: ignore[arg-type]
+            else:
+                yield self._commit_printer(
+                    element,  # type: ignore[arg-type]
+                    self._commit_id_to_branches.get(element.commit_id),  # type: ignore[union-attr]
```

Comment:
why we have these ignore comments here?

Reply:
the `element` can be `Commit` or `Draft`, but the `self._commit_printer` has argument type `commit` and only `Commit` has attribute `commit_id`

Reply:
That means the problem is caused by mypy cannot find out the relation between is_draft and the type of element? I think it can be solved by update the type hint of the `_generate_commits_and_drafts`

Reply:
Also I think it is a bad idea to use a bool to indicate the return value type.
Can we use other method to eliminate the bool like monkey patch?